imagetest/testworkflow.go

```diff
@@ -355,12 +355,12 @@ func finalizeWorkflows(ctx context.Context, tests []*TestWorkflow, zone, gcsPref
 
 		twf.wf.Zone = zone
 
-		createVMsStep := twf.wf.Steps[createVMsStepName]
-		for _, vm := range createVMsStep.CreateInstances.Instances {
-			if vm.Zone != "" && vm.Zone != twf.wf.Zone {
-				log.Printf("VM %s zone is set to %s, differing from workflow zone %s for test %s, not overriding\n", vm.Name, vm.Zone, twf.wf.Zone, twf.Name)
-			}
-			if machineType != "" {
+		if machineType != "" {
```

Comment:
This will avoid warning about mismatched zones unless the machine_type flag is set, and could still panic if the machine_type flag was set when createVMsStep is nil

Reply:
When support for machinetypes was introduced, it seems that the createVMsStep was not nil as long as machinetype was set, so I don't think that should be a major concern here as it was not broken in the past.

Another possible fix would be to investigate which calls in the networkperf test suite are causing createVmsStep to be nil @drewhli 

https://github.com/GoogleCloudPlatform/guest-test-infra/pull/390

Reply:
fixed so that the only change is to check if createVMsStep is nil before deferencing

the other logic should be the same as before for logging the zone and machine type